clang/include/clang/AST/Mangle.h

```diff
@@ -153,7 +160,8 @@ class MangleContext {
 
   void mangleObjCMethodName(const ObjCMethodDecl *MD, raw_ostream &OS,
                             bool includePrefixByte = true,
-                            bool includeCategoryNamespace = true);
+                            bool includeCategoryNamespace = true,
+                            bool isThunk = true);
```

Comment:
Compilers introduce thunks for a lot of different reasons, so I don't like having super-generic names like `isThunk` floating around. Maybe an `ObjCThunkKind` or an `isNilCheckThunk`?

Reply:
Driveby comment, feel free to ignore: 3 sequential optional boolean parameters is past the point at which you should come up with a flags enum for readability.